prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java

```diff
@@ -0,0 +1,127 @@
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		LinkedList<WorkParameter> params = new LinkedList<>();
+		params.add(WorkParameter.builder().key("url").type(WorkParameterType.TEXT).optional(false)
+				.description("URL to send request to").build());
+		params.add(WorkParameter.builder().key("method").type(WorkParameterType.TEXT).optional(false)
+				.description("The HTTP method").build());
+		params.add(WorkParameter.builder().key("content").type(WorkParameterType.TEXT).optional(true)
+				.description("The content of the HTTP request").build());
+		params.add(WorkParameter.builder().key("username").type(WorkParameterType.TEXT).optional(true)
+				.description("Username for basic HTTP authentication").build());
+		params.add(WorkParameter.builder().key("password").type(WorkParameterType.TEXT).optional(true)
+				.description("Password for basic HTTP authentication").build());
+		params.add(WorkParameter.builder().key("response-key").type(WorkParameterType.TEXT).optional(true)
+				.description("The content of the response will be stored in this key").build());
+		return params;
+	}
+
+	@Override
+	public @NonNull List<WorkFlowTaskOutput> getWorkFlowTaskOutputs() {
+		return List.of(WorkFlowTaskOutput.OTHER);
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		String url = "";
+		try {
+			url = getRequiredParameterValue(workContext, "url");
+			String method = getRequiredParameterValue(workContext, "method");
+
+			HttpMethod httpMethod = HttpMethod.valueOf(method.toUpperCase());
+
+			ResponseEntity<String> responseEntity = restService.exchange(url, httpMethod,
+					buildRequestEntity(workContext));
+
+			if (!responseEntity.getStatusCode().is2xxSuccessful()) {
+				throw new RestClientException(
+						"Request failed with HTTP status code " + responseEntity.getStatusCodeValue());
+			}
+
+			processResponseEntity(workContext, responseEntity);
+
+		}
+		catch (MissingParameterException | IllegalArgumentException e) {
+			log.error("Rest task failed for url " + url, e);
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+
+		return new DefaultWorkReport(WorkStatus.COMPLETED, workContext);
+	}
+
+	protected void processResponseEntity(WorkContext workContext, ResponseEntity<String> responseEntity)
+			throws RestClientException {
+		String responseKey = getOptionalParameterValue(workContext, "response-key", "");
+
+		if (responseKey.isEmpty()) {
+			return;
+		}
+
+		workContext.put(responseKey, responseEntity.getBody());
+	}
+
+	protected HttpEntity<String> buildRequestEntity(WorkContext workContext) throws RestClientException {
+		String content = getOptionalParameterValue(workContext, "content", "");
+		return new HttpEntity<>(content, buildHttpHeaders(workContext));
+	}
+
+	protected HttpHeaders buildHttpHeaders(WorkContext workContext) throws RestClientException {
+		HttpHeaders httpHeaders = new HttpHeaders();
+
+		httpHeaders.setContentType(MediaType.APPLICATION_JSON);
```

Comment:
should it be configurable?

Reply:
we provide basic/general/common config as a base class. user can use it as is or extend.